src/main/scala/viper/gobra/frontend/Parser.scala

```diff
@@ -1036,6 +1049,26 @@ object Parser {
         exactWord("uint64") ^^^ PUInt64Type() |
         exactWord("uintptr") ^^^ PUIntPtr()
 
+    lazy val predeclaredTypeSeparate: Parser[PPredeclaredType] =
+      exactWord("bool") ~ not("(" | ".") ^^^ PBoolType() |
+        exactWord("string") ~ not("(" | ".") ^^^ PStringType() |
+        exactWord("perm") ~ not("(" | ".") ^^^ PPermissionType() |
+        // signed integer types
+        exactWord("rune") ~ not("(" | ".") ^^^ PRune() |
+        exactWord("int") ~ not("(" | ".") ^^^ PIntType() |
+        exactWord("int8") ~ not("(" | ".") ^^^ PInt8Type() |
+        exactWord("int16") ~ not("(" | ".") ^^^ PInt16Type() |
+        exactWord("int32") ~ not("(" | ".") ^^^ PInt32Type() |
+        exactWord("int64") ~ not("(" | ".") ^^^ PInt64Type() |
+        // unsigned integer types
+        exactWord("byte") ~ not("(" | ".") ^^^ PByte() |
+        exactWord("uint") ~ not("(" | ".") ^^^ PUIntType() |
+        exactWord("uint8") ~ not("(" | ".") ^^^ PUInt8Type() |
+        exactWord("uint16") ~ not("(" | ".") ^^^ PUInt16Type() |
+        exactWord("uint32") ~ not("(" | ".") ^^^ PUInt32Type() |
+        exactWord("uint64") ~ not("(" | ".") ^^^ PUInt64Type() |
+        exactWord("uintptr") ~ not("(" | ".") ^^^ PUIntPtr()
+
```

Comment:
wouldn't the following work instead?
```scala
lazy val predeclaredTypeSeparate: Parser[PPredeclaredType] = predeclaredType <~ not("(" | ".")
```

Reply:
Yes, but I had the impression that my version has a better performance.